### PR TITLE
Revert property transition

### LINKED ISSUE #11373: API regression/breakage
since 17d91b9d4d2d208f4a847395198cdbbcad18de93, ping @christoph-heinrich 

example code from client.h:
```c
int flag = 1;
mpv_set_property(ctx, "property", MPV_FORMAT_FLAG, &flag);
```

setproperty_fn copies the bytes:
grafik

since `MPV_FORMAT_FLAG` now maps to `CONF_TYPE_BOOL` this copies 1 byte:
grafik

but an int is bigger than 1 byte, so you get mpv acting upon incorrect/unitialized data

`mpv_get_property` presumably has the same issue

### PR DESCRIPTION
**Pull request by christoph-heinrich**
Changing the CONF_TYPE_FLAG was a bad idea because mpv_node.u.flag
continues to be an int, leading to a mismatch in type sizes which can
lead to problems with memcpy().

Closes #11373

### CODE REVIEW

File: options/m_option.h [hunk 1]
```diff
@@ -212,7 +212,7 @@ struct m_sub_options {
     bool (*get_sub_options)(int index, const struct m_sub_options **sub);
 };
 
-#define CONF_TYPE_BOOL          (&m_option_type_bool)
+#define CONF_TYPE_FLAG          (&m_option_type_flag)
```
Comment (sfan5) [inline options/m_option.h:215]: I *think* you could leave the option changes, we just need `CONF_TYPE_FLAG` back for the sake of conversion

Comment (christoph-heinrich) [reply options/m_option.h:216]: done

File: options/m_property.c [hunk 2]
```diff
@@ -352,7 +352,7 @@ void m_properties_print_help_list(struct mp_log *log,
     mp_info(log, "\nTotal: %d properties\n", count);
 }
 
-int m_property_bool_ro(int action, void* arg, int var)
+int m_property_bool_ro(int action, void* arg, bool var)
```
Comment (Dudemanguy) [inline options/m_property.c:355]: Should you also change the cast below to bool?

Comment (sfan5) [reply options/m_property.c:355]: Yes, that looks like another memory error :thinking:

Comment (christoph-heinrich) [reply options/m_property.c:355]: done

**Comment by sfan5** [comment]
One last thing:
> Partially revert "options: transition properties from flag to bool"

while technically true it's just changing 3 lines around, I would use a different commit message

**Review by Dudemanguy — APPROVED** [review]
Aside from sfan's comments about the commit titles, code LGTM. I guess it might be worth introducing `MPV_FORMAT_BOOL` for libmpv later on.

File: options/m_option.c [hunk 3]
```diff
@@ -224,8 +224,70 @@ const m_option_type_t m_option_type_bool = {
...
+    return bool_get(opt, ta_parent, dst, &(bool){VAL(src)});
+}
+
+static bool flag_equal(const m_option_t *opt, void *a, void *b)
+{
+    return VAL(a) == VAL(b);
+}
+
+const m_option_type_t m_option_type_flag = {
+    // need yes or no in config files
```
Comment (sfan5) [inline options/m_option.c:274]: might also be worth pointing out somewhere that this only still exists for libmpv interopability and should not be used anywhere

Comment (christoph-heinrich) [reply options/m_option.c:275]: hope it's fine now

**Comment by Dudemanguy** [comment]
Should be `client API:` (lowercase). I'm not sure if we really have a rule for capitalization on revert commits but I guess "Partially revert" is fine since "Revert" is commonly used.

**Review by sfan5 — APPROVED** [review]
Tested & works.

File: options/m_property.c [hunk 4]
```diff
@@ -352,11 +352,11 @@ void m_properties_print_help_list(struct mp_log *log,
...
 {
     switch (action) {
     case M_PROPERTY_GET:
-        *(int *)arg = !!var;
+        *(bool *)arg = !!var;
```
Comment (N-R-K) [inline options/m_property.c:359]: minor nit: Given that `var` is now a proper `bool` the `!!` is redundant.

**Pull request merged**